irr/src/CIrrDeviceSDL.cpp

```diff
@@ -218,8 +218,79 @@ int CIrrDeviceSDL::findCharToPassToIrrlicht(int assumedChar, EKEY_CODE key)
 	case KEY_NUMLOCK:
 		return 0;
 	default:
-		return assumedChar;
+		break;
+	}
+
+	int retChar = assumedChar;
+
+	if (isShiftPressed) {
+		switch (assumedChar) {
```

Comment:
not everyone has an US keyboard layout so how will this possibly be correct?

Reply:
It looks like I had really hasted with publishing the review, especially at 3am...

The only better solution I see so far is to receive it as an `SDL_TEXTINPUT` event, and in this case we'll be able to retrieve the exact character (it should be checked it will be only available on desktop). I didn't find any better solution so far to solve the problem.

Reply:
From what I understand, earlier `SDL_keysym` provided the `unicode` field which could be used for this purpose, but now it's removed.

Reply:
> The only better solution I see so far is to receive it as an `SDL_TEXTINPUT` event, and in this case we'll be able to retrieve the exact character (it should be checked it will be only available on desktop).

Please don't. I don't think it is possible to make this work (sanely) with IMEs.

Reply:
Ok, I'm then closing the pull request because it seems like the issue cannot be solved in a "sane" way with the current system of key binding.